packages/types/src/types.ts

```diff
@@ -20,3 +24,172 @@ export enum ProducedBlockSource {
+    SignedBeaconBlockHeader: phase0.SignedBeaconBlockHeader;
+    BeaconBlock: deneb.BeaconBlock;
+    BeaconBlockBody: deneb.BeaconBlockBody;
+    BeaconState: deneb.BeaconState;
+    SignedBeaconBlock: deneb.SignedBeaconBlock;
+    Metadata: altair.Metadata;
+    LightClientHeader: deneb.LightClientHeader;
+    LightClientBootstrap: deneb.LightClientBootstrap;
+    LightClientUpdate: deneb.LightClientUpdate;
+    LightClientFinalityUpdate: deneb.LightClientFinalityUpdate;
+    LightClientOptimisticUpdate: deneb.LightClientOptimisticUpdate;
+    LightClientStore: deneb.LightClientStore;
+    BlindedBeaconBlock: deneb.BlindedBeaconBlock;
+    BlindedBeaconBlockBody: deneb.BlindedBeaconBlockBody;
+    SignedBlindedBeaconBlock: deneb.SignedBlindedBeaconBlock;
+    ExecutionPayload: deneb.ExecutionPayload;
+    ExecutionPayloadHeader: deneb.ExecutionPayloadHeader;
+    BuilderBid: deneb.BuilderBid;
+    SignedBuilderBid: deneb.SignedBuilderBid;
+    SSEPayloadAttributes: deneb.SSEPayloadAttributes;
+    BlockContents: {block: BeaconBlock<ForkBlobs>; kzgProofs: deneb.KZGProofs; blobs: deneb.Blobs};
+    SignedBlockContents: {
+      signedBlock: SignedBeaconBlock<ForkBlobs>;
+      kzgProofs: deneb.KZGProofs;
+      blobs: deneb.Blobs;
+    };
+    ExecutionPayloadAndBlobsBundle: deneb.ExecutionPayloadAndBlobsBundle;
+    BlobsBundle: deneb.BlobsBundle;
+    Contents: deneb.Contents;
+    SyncCommittee: altair.SyncCommittee;
+    SyncAggregate: altair.SyncAggregate;
+  };
+};
+
+export type TypesFor<F extends ForkName, K extends keyof TypesByFork[F] | void = void> = K extends void
+  ? TypesByFork[F]
+  : TypesByFork[F][Exclude<K, void>];
+
+export type BeaconBlockHeader<F extends ForkAll = ForkAll> = TypesByFork[F]["BeaconBlockHeader"];
+export type SignedBeaconBlockHeader<F extends ForkAll = ForkAll> = TypesByFork[F]["SignedBeaconBlockHeader"];
+
+export type BeaconBlock<F extends ForkAll = ForkAll> = TypesByFork[F]["BeaconBlock"];
+export type BlindedBeaconBlock<F extends ForkExecution = ForkExecution> = TypesByFork[F]["BlindedBeaconBlock"];
+
+export type SignedBeaconBlock<F extends ForkAll = ForkAll> = TypesByFork[F]["SignedBeaconBlock"];
+export type SignedBlindedBeaconBlock<F extends ForkExecution = ForkExecution> =
+  TypesByFork[F]["SignedBlindedBeaconBlock"];
+
+export type BeaconBlockBody<F extends ForkAll = ForkAll> = TypesByFork[F]["BeaconBlockBody"];
+export type BlindedBeaconBlockBody<F extends ForkExecution = ForkExecution> = TypesByFork[F]["BlindedBeaconBlockBody"];
+
+export type BlockContents<F extends ForkBlobs = ForkBlobs> = TypesByFork[F]["BlockContents"];
+export type SignedBlockContents<F extends ForkBlobs = ForkBlobs> = TypesByFork[F]["SignedBlockContents"];
+export type SignedOrUnsignedBlockContents<F extends ForkBlobs = ForkBlobs> = BlockContents<F> | SignedBlockContents<F>;
+
+export type BeaconBlockOrContents<FB extends ForkAll = ForkAll, FC extends ForkBlobs = ForkBlobs> =
```

Comment:
```suggestion
export type BeaconBlockOrContents<FB extends PreForkBlobs = ForkAll, FC extends ForkBlobs = ForkBlobs> =
```

to make it more tighter